Source/WebInspectorUI/UserInterface/Base/Main.js

```diff
@@ -237,6 +237,12 @@ WI.contentLoaded = function()
     WI.settings.resourceCachingDisabled.addEventListener(WI.Setting.Event.Changed, WI._resourceCachingDisabledSettingChanged, WI);
     WI.settings.experimentalAllowInspectingInspector.addEventListener(WI.Setting.Event.Changed, WI._allowInspectingInspectorSettingChanged, WI);
 
+    if (WI.engineeringSettingsAllowed()) {
+        WI.settings.engineeringShowSiteIsolationDebugOverlay.addEventListener(WI.Setting.Event.Changed, () => {
+            WI.mainTarget.PageAgent.overrideSetting(InspectorBackend.Enum.Page.Setting.SiteIsolationDebugOverlay, WI.settings.engineeringShowSiteIsolationDebugOverlay.value);
+        });
```

Comment:
since Web Inspector uses its own event listener system, we try not to use arrow functions as they can keep more things alive than intended
```suggestion
        WI.settings.engineeringShowSiteIsolationDebugOverlay.addEventListener(WI.Setting.Event.Changed, function(event) {
            WI.mainTarget.PageAgent.overrideSetting(InspectorBackend.Enum.Page.Setting.SiteIsolationDebugOverlay, WI.settings.engineeringShowSiteIsolationDebugOverlay.value);
        });
}, WI.settings.engineeringShowSiteIsolationDebugOverlay);
```